### PR TITLE
consolidated wire+assign to just wire, with expression inlined

### PR DESCRIPTION
**Pull request by vmurali**
### Contributor Checklist

- [ ] Did you add Scaladoc to every public function/method?
- [ ] Did you update the FIRRTL spec to include every new feature/behavior?
- [x] Did you add at least one test demonstrating the PR?
- [ ] Did you delete any extraneous printlns/debugging code?
- [ ] Did you specify the type of improvement?
- [ ] Did you state the API impact?
- [x] Did you specify the code generation impact?
- [ ] Did you request a desired merge strategy?
- [ ] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

- backend code generation

#### Backend Code Generation Impact

 Yes

"
wire logic[...] x;
assign x = expr;
"
becomes
wire logic[...] x = expr;

### Reviewer Checklist (only modified by reviewer)
- [x] Did you add the appropriate labels?
- [x] Did you mark the proper milestone (1.2.x, 1.3.0, 1.4.0) ?
- [x] Did you review?
- [x] Did you check whether all relevant Contributor checkboxes have been checked?
- [x] Did you mark as `Please Merge`?

### CODE REVIEW

File: src/main/scala/firrtl/Emitter.scala [hunk 1]
```diff
@@ -874,8 +874,8 @@ class VerilogEmitter extends SeqTransform with Emitter {
           if (!options.disableRandomization)
             initialize(e, sx.reset, sx.init)
         case sx: DefNode =>
-          declare("wire", sx.name, sx.value.tpe, sx.info)
-          assign(WRef(sx.name, sx.value.tpe, NodeKind, SourceFlow), sx.value, sx.info)
+          declare("wire", sx.name, sx.value.tpe, sx.info, sx.value)
+          // assign(WRef(sx.name, sx.value.tpe, NodeKind, SourceFlow), sx.value, sx.info)
```
Comment (albert-magyar) [inline src/main/scala/firrtl/Emitter.scala:878]:
```suggestion
```

**Comment by albert-magyar** [comment]
I marked it as 1.2.x -- it may not backport cleanly, since the signature for declare was changed for #1050, but I can manually backport it all the way, since it's a worthwhile change.

**Comment by jackkoenig** [comment]
Because https://github.com/freechipsproject/firrtl/pull/1595 is not yet merged both backports have to be manual

**Comment by albert-magyar** [comment]
backport 1.3.x

**Comment by albert-magyar** [comment]
@Mergifyio backport 1.3.x

**Pull request merged**